main/command/src/main/scala/sbt/MainLoop.scala

```diff
@@ -65,6 +65,10 @@ object MainLoop {
       val newLogging = state.globalLogging.newLogger(out, logBacking)
       transferLevels(state, newLogging)
       val loggedState = state.copy(globalLogging = newLogging)
+      def isInteractive = System.console() != null
+      def hasShell = state.remainingCommands contains "shell"
+      if (isInteractive && !hasShell)
+        state.log info "!!! Executing in batch mode !!! For better performance, hit [ENTER] to remain in the sbt shell"
```

Comment:
I'm not so sure if warning is the right level here. sbt is able to reduce the log level using `--warn` (This could get rid of Resolving etc.), and I feel like "tip of the day" message sits squarely on the "info" level. Yes, your build is going to compile slower, but it's not like it's destroying your code.